eolearn/core/eodata.py

```diff
@@ -750,6 +751,43 @@ def consolidate_timestamps(self, timestamps: list[dt.datetime]) -> set[dt.dateti
 
         return remove_from_patch
 
+    def temporal_subset(
+        self, timestamps: Iterable[dt.datetime] | Iterable[int] | Callable[[dt.datetime], bool]
+    ) -> EOPatch:
+        """Returns an EOPatch that only contains data for the temporal subset defined by `timestamps`.
+
+        For array based data appropriate temporal slices are extracted. For vector data a filtration is performed.
+
+        :param timestamps: Parameter that defines the temporal subset. Can be a collection of timestamps, a
+            collection of timestamp indices, or a callable that returns whether a timestamp should be kept.
+        """
+        timestamp_indices = self._parse_temporal_subset_input(timestamps)
+        new_timestamps = [ts for i, ts in enumerate(self.get_timestamps()) if i in set(timestamp_indices)]
```

Comment:
is the set here necessary? I don't expect duplicates, but if it makes inference faster, then it's fine

Reply:
i wanted to see what the impact is. For a whole year worth of indices, if you use a set instead of a list it's about one magnitude faster (still under 1ms in both cases). However here you must also BUILD the set. Turns out constructing the set is slower than the suboptimal list lookup, which sounded weird to me.

But then i remembered that this is Python and Python is unpredictable in best of times.
![image](https://github.com/sentinel-hub/eo-learn/assets/31988337/a09e0d1d-c468-4619-9990-ea8530492a13)

The set is recalculated every time, and it should be. But one quickly forgets that with list comprehensions.

Reply:
i removed the set transformation since even for a full year the whole thing is under ms when using lists

Reply:
just a random question, if you know by heart. Would the set also be recalculated every time if this was a standard for loop?

Reply:
yes, because one can do some stupid things like
```
for x in blabla:
    if x in fetch_constantly_updating_thing():
        ...
```
So it was a problem of my brain assuming stuff